### PR TITLE
ci(publish-nx.yml): fix oss publish

### PR DESCRIPTION
**Pull request by ACoolmanTelicent**
See https://github.com/telicent-oss/rdf-libraries/compare/main...ci(publish-nx.yml)/ci-fix-with-req-changes

### CODE REVIEW

File: .github/workflows/publish-nx.yml [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
     paths:
       - CHANGELOG.md
 
 jobs:
   main:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+      packages: write
```
Comment (Greg-teli100) [inline .github/workflows/publish-nx.yml:16]: Is this required? I don't think we are using github packages.
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#visibility-and-access-permissions-for-packages
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages#access-tokens

If we are, is read on `packages` enough?

Comment (ACoolmanTelicent) [reply .github/workflows/publish-nx.yml:16]: Ah, probably not. This is probably just an artifact of my "solution shotgun" fixing.

Comment (Greg-teli100) [reply .github/workflows/publish-nx.yml:16]: According to https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry your original config should work. 
I note this is using v4 of the action though which could be using a newer version of npm in some part. Npm login defaults were changed were made between node 16 and node 20 .

https://github.com/actions/setup-node/compare/v3...v4#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6

https://docs.npmjs.com/accessing-npm-using-2fa#sign-in-from-the-command-line-using---auth-typeweb

Comment (ACoolmanTelicent) [reply .github/workflows/publish-nx.yml:16]: My original do you mean, the base? It definitely doesn't work. I'll try v4

Comment (ACoolmanTelicent) [reply .github/workflows/publish-nx.yml:16]: v4 Doesn't allow me to remove "Configure npm" step:
https://github.com/telicent-oss/rdf-libraries/actions/runs/8139202853


I'll remove the `packages: write` in a new PR, and then I'll leave this for now as this workflow works, and I need to check the other 6 or so workflows

**Comment by ACoolmanTelicent** [comment]
New PR https://github.com/telicent-oss/rdf-libraries/pull/16

**Pull request closed without merging**